src/app/api/users/route.ts

```diff
@@ -0,0 +1,29 @@
+import { createUser } from '@/server/actions/users';
+import { zCreateUserRequest } from '@/types/persons';
+import { NextRequest, NextResponse } from 'next/server';
+
+export async function POST(request: NextRequest) {
+  try {
+    const body = await request.json();
+
+    const validateResult = zCreateUserRequest.safeParse(body);
+    if (!validateResult.success) {
+      return NextResponse.json(
+        { message: validateResult.error },
+        { status: 400 }
+      );
+    }
+
+    const result = await createUser(validateResult.data);
+    if (!result) {
```

Comment:
I don't think that this is necessary. I realized that this was just an issue of copying code without fully understanding. If you think otherwise feel free to keep it but it shouldn't be necessary